cmd/builder/main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	bolt "github.com/johnnadratowski/golang-neo4j-bolt-driver"
+	"log"
+	"time"
+	"fmt"
+	"flag"
+)
+
+var codeListID = flag.String("code-list-id", "e44de4c4-d39e-4e2f-942b-3ca10584d078", "")
+var instanceID = flag.String("instance-id", "12345", "")
+var dimensionName = flag.String("dimension-name", "Aggregate", "")
+
+func main() {
+
+	driver := bolt.NewDriver()
+
+	connection, err := driver.OpenNeo("bolt://localhost:7687")
+	if err != nil {
+		return
+	}
+	defer connection.Close()
+
+	createInstanceHierarcyConstraints(connection)
+	if err != nil {
+		log.Println(err.Error())
+		return
+	}
+
+	err = cloneNodes(connection)
+	if err != nil {
+		log.Println(err.Error())
+	}
+
+	err = cloneRelationships(connection)
+	if err != nil {
+		log.Println(err.Error())
+	}
+
+	// mark nodes that have data
+	err = setHasData(connection)
+	if err != nil {
+		log.Println(err.Error())
+	}
+
+	// prune branches with no data
+
+	// set the number of children
+	err = setNumberOfChildren(connection)
+	if err != nil {
+		log.Println(err.Error())
+	}
+
+}
+func createInstanceHierarcyConstraints(connection bolt.Conn) error {
+	// CREATE CONSTRAINT ON (d:ASHE07E_Dimension6) ASSERT d.value IS UNIQUE;
+	createConstraint := fmt.Sprintf("CREATE CONSTRAINT ON (n:`_hierarchy_node_%s_%s`) ASSERT n.code IS UNIQUE;", *instanceID, *dimensionName)
+	stmtInsert, err := connection.PrepareNeo(createConstraint)
+	if err != nil {
+		return err
+	}
+	results, err := stmtInsert.ExecNeo(nil)
+	if err != nil {
+		return err
+	}
+	fmt.Println(results)
```

Comment:
Any reason why this is using `fmt` instead of `log`

Reply:
Changed to use log